test/interactions/clickInteractionTests.ts

```diff
@@ -2,156 +2,186 @@
       });
+    });
 
-      it("does not trigger callback when touch event is cancelled", () => {
-        let callbackCalled = false;
-        let callback = () => callbackCalled = true;
-        clickInteraction.onClick(callback);
-
-        TestMethods.triggerFakeTouchEvent("touchstart", component.content(), [{x: SVG_WIDTH / 2, y: SVG_HEIGHT / 2}]);
-        TestMethods.triggerFakeTouchEvent("touchcancel", component.content(), [{x: SVG_WIDTH / 2, y: SVG_HEIGHT / 2}]);
-        TestMethods.triggerFakeTouchEvent("touchend", component.content(), [{x: SVG_WIDTH / 2, y: SVG_HEIGHT / 2}]);
-        assert.isFalse(callbackCalled, "callback not called since click was interrupted");
-
-        svg.remove();
+    [TestMethods.InteractionMode.Mouse, TestMethods.InteractionMode.Touch].forEach((mode) => {
+      describe(`invoking callbacks with ${TestMethods.InteractionMode[mode]} events`, () => {
+        let callback: ClickTestCallback;
+
+        beforeEach(() => {
+          callback = makeClickCallback();
+          clickInteraction.onClick(callback);
+        });
+
+        it("invokes onClick callback on single location click", () => {
+          const point = {x: SVG_WIDTH / 2, y: SVG_HEIGHT / 2};
+
+          clickPoint(point, mode);
+
+          assert.isTrue(callback.called, "callback called on clicking Component without moving pointer");
+          assert.deepEqual(callback.lastPoint, point, "was passed correct point");
+        });
+
+        it("provides correct release point to onClick callback on click", () => {
+          const startPoint = {x: SVG_WIDTH / 2, y: SVG_HEIGHT / 2};
+          const endPoint = {x: SVG_WIDTH / 4, y: SVG_HEIGHT / 4};
+
+          clickPointWithMove(startPoint, endPoint, mode);
+
+          assert.isTrue(callback.called, "callback called on clicking and releasing inside the Component");
+          assert.deepEqual(callback.lastPoint, endPoint, "was passed mouseup point");
+        });
+
+        it("does not invoke callback if click is released outside Component", () => {
+          const startPoint = {x: SVG_WIDTH / 2, y: SVG_HEIGHT / 2};
+          const endPoint = {x: SVG_WIDTH * 2, y: SVG_HEIGHT * 2};
+
+          clickPointWithMove(startPoint, endPoint, mode);
+
+          assert.isFalse(callback.called, "callback not called if click is released outside Component");
+        });
+
+        it("does not invoke callback if click is started outside Component", () => {
+          const startPoint = {x: SVG_WIDTH * 2, y: SVG_HEIGHT * 2};
+          const endPoint = {x: SVG_WIDTH / 2, y: SVG_HEIGHT / 2};
+
+          clickPointWithMove(startPoint, endPoint, mode);
+
+          assert.isFalse(callback.called, "callback not called if click was started outside Component");
+        });
+
+        it("invokes callback if the pointer is moved out then back inside the Component before releasing", () => {
+          TestMethods.triggerFakeInteractionEvent(mode,
```

Comment:
do you think it is possible to factor this into `clickPointWithMove` with additional parameter?